clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -198,6 +208,41 @@ public boolean hasCompletedFetches() {
         return !completedFetches.isEmpty();
     }
 
+    /**
+     * Return if there are buffered CompletedFetches for a paused TopicPartition.
+     *
+     * @param tp {@link TopicPartition}
+     * @return  {@code true} if there are buffered {@link CompletedFetch},  {@false} otherwise
+     */
+    public boolean hasPausedCompletedFetchesFor(TopicPartition tp) {
+        return pausedCompletedFetchesPerTopicPartition.containsKey(tp);
+    }
+
+    /**
+     * Add the TopicPartition to the list of recently unpaused partitions.
+     * @param tp {@link TopicPartition}
+     */
+    public void addRecentlyUnpausedPartition(TopicPartition tp) {
+        this.recentlyUnPausedTopicPartitions.add(tp);
+    }
+
+    /**
+     * Remove the TopicPartition from list of recently unpaused partitions.
+     * @param tp {@link TopicPartition}
+     */
+    public void removeRecentlyUnpausedPartition(TopicPartition tp) {
+        this.recentlyUnPausedTopicPartitions.remove(tp);
+    }
+
+    /**
+     * Clear the data for all paused partitions.
+     */
+    public void clearBufferedDataForPausedPartitions() {
```

Comment:
Can we re-use the existing method `clearBufferedDataForUnassignedPartitions()` and clear data as necessary in that method?